### PR TITLE
Handle starting a call while listen to a broadcast

### PR DESCRIPTION
**Pull request by weeman1337**
Part of https://github.com/vector-im/element-web/issues/23282

- If listening to a broadcast, pause
- If recording a broadcast, show dialog, don't start call

Other cases are out of scope for this PR

call_vb

PSF-1716

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/LegacyCallHandler.tsx [hunk 1]
```diff
@@ -932,6 +934,15 @@ export default class LegacyCallHandler extends EventEmitter {
     }
 
     public async placeCall(roomId: string, type?: CallType, transferee?: MatrixCall): Promise<void> {
+        // Pause current broadcast, if any
+        SdkContextClass.instance.voiceBroadcastPlaybacksStore.getCurrent()?.pause();
+
+        if (SdkContextClass.instance.voiceBroadcastRecordingsStore.getCurrent()) {
+            // Do not start a calc, if recording a broadcast
```
Comment (kerryarchibald) [inline src/LegacyCallHandler.tsx:941]: Typo `Do not start a call*`

**Review by kerryarchibald — COMMENTED** [review]
Tested on the netlify - you can still:
1. start recording a voice broadcast and then answer an incoming call
2. start a call, then start recording a voice broadcast
It feels like for completeness these cases should be covered with the 'can't start a call' dialog too.

Also if I start a call in roomA, then try to start a broadcast in roomB it seems like nothing happens. After ending the call the broadcast dialog from roomB is visible - so maybe the sticky elements are just stacking.

**Comment by weeman1337** [comment]
Thanks for testing in detail @kerryarchibald :+1: I will take your feedback into account in our further planning.

This PR only covers the case if you try to start a call if you are currently listening to or recording a voice broadcast.

**Review by kerryarchibald — APPROVED** [review]
LGTM

**Pull request merged**